src/state-manager/index.js

```diff
@@ -0,0 +1,38 @@
+// @flow
```

Comment:
Is this abstraction necessary? Isn't it sufficient for the consumers to pull in the exact state manager(s) they want to use, and require all state managers to have implementations of `updateStep` and `getActiveStep`? No blocker for merge, I can propose a solution in a PR! :raised_hands:

Reply:
Got you! Thanks for the spot :) No need for such an abstraction since we have flow to enforce only appropriate methods.